torch/_dynamo/convert_frame.py

```diff
@@ -260,15 +260,20 @@ def _convert_frame_assert(
             recompiles_log.isEnabledFor(logging.DEBUG) or config.error_on_recompile
         ):
             if is_guard_failure_reporting_enabled():
-                message = (
-                    f"Recompiling function {code.co_name} in {code.co_filename}:{code.co_firstlineno}",
-                    f"triggered by the following guard failure: {str(guard_failures[code][-1])}",
+                failures = str(guard_failures[code][-1])
+                if config.report_all_guard_failures:
+                    failures = failures.strip().split("\n")  # type: ignore[assignment]
+                guard_failure_details = (
+                    f"triggered by the following guard failure(s): {failures}"
                 )
             else:
-                message = (
-                    f"Recompiling function {code.co_name} in {code.co_filename}:{code.co_firstlineno}",
-                    "set env var TORCHDYNAMO_REPORT_GUARD_FAILURES=1 to debug further",
+                guard_failure_details = (
+                    "set env var TORCHDYNAMO_REPORT_GUARD_FAILURES=1 to debug further"
                 )
+            message = (
+                f"Recompiling function {code.co_name} in {code.co_filename}:{code.co_firstlineno}",
```

Comment:
preexisting issue (no need to act on it). This is creating a tuple of strings, rather than concatenating them lol

Reply:
Maybe I'll do so in follow up PR actually, the formatting rn is pretty horrid lol